enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.unit.spec.tsx

```diff
@@ -0,0 +1,248 @@
+} from "__support__/server-mocks";
+import { mockSettings } from "__support__/settings";
+import { renderWithProviders, screen } from "__support__/ui";
+import { useInitData } from "embedding-sdk/hooks";
+import { SDK_REDUCERS, useSdkSelector } from "embedding-sdk/store";
+import { getIsLoggedIn, getLoginStatus } from "embedding-sdk/store/selectors";
+import type { LoginStatusError } from "embedding-sdk/store/types";
+import { createMockConfig } from "embedding-sdk/test/mocks/config";
+import type { SDKConfigType } from "embedding-sdk/types";
+import {
+  createMockSettingDefinition,
+  createMockSettings,
+  createMockTokenFeatures,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import { createMockState } from "metabase-types/store/mocks";
+
+const TEST_USER = createMockUser();
+
+const TestComponent = ({
+  authType,
+  ...opts
+}: {
+  authType: SDKConfigType["authType"] | "invalid";
+  opts?: Partial<SDKConfigType>;
+}) => {
+  const config = createMockConfig({ authType } as Partial<SDKConfigType>);
+
+  const loginStatus = useSdkSelector(getLoginStatus);
+  const isLoggedIn = useSdkSelector(getIsLoggedIn);
+
+  useInitData({
+    config: {
+      ...config,
+      metabaseInstanceUrl: "http://localhost",
+      ...opts,
+    } as SDKConfigType,
+  });
+
+  return (
+    <div
+      data-testid="test-component"
+      data-is-logged-in={isLoggedIn}
+      data-login-status={loginStatus.status}
+      data-error-message={(loginStatus as LoginStatusError).error?.message}
+    >
+      Test Component
+    </div>
+  );
+};
+
+jest.mock("metabase/visualizations/register", () => jest.fn(() => {}));
+
+const setup = ({
+  authType,
+  isValidAuthentication = true,
+  ...configOpts
+}: Partial<Omit<SDKConfigType, "authType">> & {
+  authType: SDKConfigType["authType"] | "invalid";
+  isValidAuthentication?: boolean;
+}) => {
+  // Mock JWT SSO Endpoint
+  fetchMock.mock("http://TEST_URI/sso/metabase", {
+    id: "TEST_JWT_TOKEN",
+    exp: 1965805007,
+    iat: 1965805007,
+  });
+
+  const currentUser = TEST_USER;
+
+  setupCurrentUserEndpoint(
+    TEST_USER,
+    isValidAuthentication
+      ? undefined
+      : {
+          response: 500,
+        },
+  );
+
+  const settingValues = createMockSettings();
+  const tokenFeatures = createMockTokenFeatures();
+  const settings = [
+    createMockSettingDefinition({
+      key: "token-features",
+      value: tokenFeatures,
+    }),
+  ];
+
+  const settingValuesWithToken = {
+    ...settingValues,
+    "token-features": tokenFeatures,
+  };
+  const state = createMockState({
+    settings: mockSettings(settingValuesWithToken),
+    currentUser,
+  });
+
+  setupEnterprisePlugins();
+  setupApiKeyEndpoints([]);
+  setupSettingsEndpoints(settings);
+  setupPropertiesEndpoints(settingValuesWithToken);
+
+  renderWithProviders(<TestComponent authType={authType} {...configOpts} />, {
+    storeInitialState: state,
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
I'm not happy with this eslint, but it seems like the types for this are incredibly convoluted and would take more time than it's worth to get the types to work for this

Reply:
Just like `PublicComponentWrapper`, I don't use `mode:sdk` here because the `SdkWrapper` in `renderWithProviders` initializes the whole SDK state and calls `use-init-data` on its own. I wanted to isolate `use-init-data` and test it on its own, which is why I'm using the default mode here.